consensus/src/block_storage/sync_manager.rs

```diff
@@ -423,25 +423,19 @@ impl BlockRetriever {
             "Retrieving {} blocks starting from {}",
             num_blocks, block_id
         );
-        let mut attempt = 0_u32;
         let mut progress = 0;
         let mut last_block_id = block_id;
         let mut result_blocks: Vec<Block> = vec![];
         let mut retrieve_batch_size = MAX_BLOCKS_PER_REQUEST;
         if peers.is_empty() {
-            bail!(
-                "Failed to fetch block {} in {} attempts: no more peers available",
-                block_id,
-                attempt
-            );
+            bail!("Failed to fetch block {}: no peers available", block_id);
         }
-        let mut peer = self.pick_peer(attempt, peers);
+        let mut peer = self.pick_peer(0, peers);
+        let mut attempt = 1_u32;
```

Comment:
nit: let's rename this to `failed_attempt` for clarity and also this should be initialized to 0 instead of 1. You can modify the check  to `failed_attempt >= 4` in that case.